src/generation/generate.rs

```diff
@@ -7118,6 +7146,22 @@ where
       }
     }
   }
+
+  fn only_single_item_and_no_comments(nodes: &[Node], program: &Program) -> bool {
+    if nodes.len() != 1 {
+      return false;
+    }
+    // check for leading or trailing comments on the only child node
+    let child = nodes[0];
+    if !child.leading_comments_fast(program).is_empty() || !child.trailing_comments_fast(program).is_empty() {
+      return false;
+    }
+    // search after the trailing comma if it exists
+    match child.next_token_fast(program) {
+      Some(TokenAndSpan { token: Token::Comma, span, .. }) => span.trailing_comments_fast(program).is_empty(),
+      _ => true,
+    }
```

Comment:
The comments are stored based on tokens. So here we can check if the token after the node is a comma. If it is, then we check for its trailing comments as well.

Reply:
Ah thanks! I assumed all comments could be obtained as leading / trailing comments from the child.

As a new contributor I found it confusing trying to obtain comments. It would be great to have helper methods that obtain all the comments in this fashion :))

Reply:
It would get very busy. This is using swc, which has a steep learning curve, but I added a lot of helper methods on top of it here https://github.com/dprint/dprint-swc-ext (the code here used to be even more confusing). For using swc, if you go here... https://play.swc.rs/?version=1.3.31&code=H4sIAAAAAAAAAytJLS7RKAESOpoA%2FsNEnwsAAAA%3D&config=H4sIAAAAAAAAAy2NsQrDMAxE%2F0VzhrZDoV5bsvUjRCIXFcc2OhsaQv69dsh2nN47bfTFRG6jzAaxnrDGwj9yVNYsmExzoYEKWlWsyt4y20dKIwS3y%2FXRr8YRPtnSByrkJV6jjMmegYFRJcw49eYvaa5BOtpfHDt36r1G9Ss5zwEykOJ9gof4B7AXfpKrAAAA ...there is a `View: "Compiled Code"` drop down on the right. If you change that to "AST" it will show all the nodes. That can be helpful sometimes. Ex. In this case the comma appears after the node's end, so it's not part of the node and the node's trailing comments won't include what's after the comma.